packages/analytics/src/client.ts

```diff
@@ -0,0 +1,36 @@
+import { AnalyticsClientConfig, AnalyticsClientInterface, EventProperties, JsonMap } from './types';
+
+export type AnalyticsClientOptions = Pick<
+  AnalyticsClientConfig,
+  'defaultProperties' | 'defaultTraits'
+>;
+
+export class AnalyticsClient<T extends AnalyticsClientInterface> {
```

Comment:
Nothing inherently wrong using a class here, but it would more inline with other packages we have to use a more functional style

```ts
export function createAnalyticsClient<T extends AnalyticsClientInterface>(
  client: T,
  options: AnalyticsClientOptions
) {
  return {
    async track<K extends keyof EventProperties>(event: K, properties: EventProperties[K]) {
      return client.track(event, { ...properties, ...options.defaultProperties });
    },

    async untypedTrack(event: string, properties?: JsonMap) {
      return client.track(event, { ...properties, ...options.defaultProperties });
    },

    async screen(name: string, properties?: JsonMap) {
      return client.screen(name, { ...properties, ...options.defaultProperties });
    },

    async group(groupId: string, traits?: JsonMap) {
      return client.group(groupId, { ...traits, ...options.defaultTraits });
    },

    async identify(userId: string, traits?: JsonMap) {
      return client.identify(userId, { ...traits, ...options.defaultTraits });
    }
  };
}
```

Reply:
Right @fbwoolf did mention that - I will update.